src/mono/mono/eglib/gfile.c

```diff
@@ -155,3 +155,63 @@ g_file_set_contents (const gchar *filename, const gchar *contents, gssize length
 
 	return TRUE;
 }
+
+FILE*
+g_fopen (const gchar *path, const gchar *mode)
+{
+	FILE *fp;
+
+#ifdef HOST_WIN32
+	gunichar2 *wPath = g_utf8_to_utf16 (path, -1, 0, 0, 0);
```

Comment:
One minor optimization that Johan suggested and I wasn't able to do it correctly is that if path only contains ascii characters, it could simply use `fopen`. I tried adding the optimization [here](https://github.com/dotnet/runtime/pull/90436/commits/ab352d4e322531705781727242881989648a4f74). But it failed the test on CI. I haven't figured out how to test it locally. So I took it out. If you could get it right, that would be great!

Reply:
I don't see this optimization having any impact in the real world, especially since we are dealing with file access which is bound to be quite slow. Unless the encoding conversion is craazy slow, which I doubt.

Reply:
I don't really have any real world experience with the relevant API's. By looking at the code, we could avoid two calculations of `g_utf8_to_utf16`.
@lateralusX Do you have any insights here?

Reply:
I have not done any measurements on the speed of encoding conversions on any platform and in relation to file IO its probably a small amount of time. The idea was to avoid doing the heap allocation and encoding conversion when not really needed (all characters in ascii format so standard function can be called) and since we have the logic put into a couple of g_ functions, the optimization will be isolated, simple and straightforward. I still think avoiding string encoding conversions and heap allocations when not needed is worthwhile, but I agree that in this case the frequency of calls coming into these functions will be low.

Reply:
@lateralusX I added the ascii string check to avoid utf16 conversion. Could you do a final review on this PR ?